spirv_msl.cpp

```diff
@@ -12548,8 +12629,22 @@ string CompilerMSL::type_to_glsl(const SPIRType &type, uint32_t id)
 	// Pointer?
 	if (type.pointer)
 	{
+		assert(type.pointer_depth > 0);
+
 		const char *restrict_kw;
-		type_name = join(get_type_address_space(type, id), " ", type_to_glsl(get<SPIRType>(type.parent_type), id));
+
+		auto type_address_space = get_type_address_space(type, id);
+		auto type_decl = type_to_glsl(get<SPIRType>(type.parent_type), id);
+
+		// Work around C pointer qualifier rules. If glsl_type is a pointer type as well
+		// we'll need to emit the address space to the right.
+		// We could always go this route, but it makes the code unnatural.
+		// Prefer emitting thread T *foo over T thread* foo since it's more readable,
+		// but we'll have to emit thread T * thread * T constant bar; for example.
```

Comment:
It would make the implementation 10x smaller to never do the previous ordering, but since you've gone through the trouble, might as well *shrug*.

Reply:
Main reason is maintenance, I don't want to review 1000+ reference file changes, and I find `device T *foo` 100x more readable than `T device * foo` personally.

Reply:
Thanks, that clears it up for me, I missed the test files.